File: Class/Class.js
class Dog {
  constructor(name, age) {
    this.name = name;
    this.age = age;
  }
  dogSound() {
    console.log(this.name, "sound is barks");
  }
}

let dog = new Dog("buddy", 4);
console.log(dog); // Dog { name: 'buddy', age: 4 }
console.log(dog.name); // buddy

class Cat extends Dog {
  catSound() {
    console.log(this.name, "sound is purr");
  }
}

let cat = new Cat("Tom", 2);
console.log(cat); // Cat { name: 'Tom', age: 2}
console.log(cat.name); // Tom
cat.catSound();
cat.dogSound();

File: Class/Class.py
class Dog:
  def __init__(self, name, age):
    self.name = name
    self.age = age

dog = Dog("Buddy", 3)
print(dog) # <__main__.Dog object at 0x0000023B4A6FB2D0>
print(dog.name) # 

class Cat(Dog):
  def catSound(self):
    print(self.name, "sound is purrs")

cat = Cat("Tom", 2)
print(cat) # <__main__.Cat object at 0x
